ddtrace/tracer/tracer.go

```diff
@@ -498,3 +505,44 @@ func (t *tracer) sample(span *span) {
 	}
 	t.prioritySampling.apply(span)
 }
+
+// Start AppSec when DD_APPSEC_ENABLED is true.
+func startAppSec(t *tracer) {
+	if appsecEnabled := os.Getenv("DD_APPSEC_ENABLED"); appsecEnabled == "" {
+		return
+	} else if enabled, err := strconv.ParseBool(appsecEnabled); err != nil {
+		log.Error("appsec: could not parse DD_APPSEC_ENABLED value `%s` as a boolean value", appsecEnabled)
+		return
+	} else if !enabled {
+		return
+	}
+
+	appsecAgent, err := appsec.NewAgent(
+		t.config.httpClient,
+		appsecLogger{},
+		&appsec.Config{
+			Version:  version.Tag,
+			AgentURL: fmt.Sprintf("http://%s/", t.config.agentAddr),
```

Comment:
Is that `AgentURL` always correct @knusbaum ?

Reply:
It is not correct in the case of lambda tracing (although AppSec likely needs to be disabled or modified for lambda tracing), and will also be incorrect in the case of UDS (see: #985)

I'm starting to turn in favor of having an `agent` package that handles this, so we don't need to re-implement the agent transport in AppSec, the profiler, tracer and whatever else.

Reply:
> I'm starting to turn in favor of having an agent package that handles this, so we don't need to re-implement the agent transport in AppSec, the profiler, tracer and whatever else.

It would be great indeed.
I saw the profiler does the same thing as I do here.

Can we create this new package in a separate PR and keep this for now?